src/java.base/share/classes/java/nio/X-Buffer.java.template

```diff
@@ -2045,8 +2045,17 @@ public abstract sealed class $Type$Buffer
      */
     public $Type$Buffer append(CharSequence csq, int start, int end) {
         if (csq instanceof CharBuffer cb) {
-            int pos = position();
+            //
+            // pre-emptively check for overflow as put(int,CharBuffer,int,int)
+            // would throw IndexOutOfBoundsException instead
```

Comment:
The comment is confusing, I think you just want to say that the append method throws BufferOverflowException when there is insufficient space in the buffer.

Reply:
Comment updated and test code moved in 118076c.